pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -0,0 +1,100 @@
+package v0alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Repository struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec RepositorySpec `json:"spec,omitempty"`
+}
+
+type LocalRepositoryConfig struct {
+	Path string `json:"path,omitempty"`
+}
+
+type S3RepositoryConfig struct {
+	Region string `json:"region,omitempty"`
+	Bucket string `json:"bucket,omitempty"`
+
+	// TODO: Add ACL?
+	// TODO: Encryption??
+	// TODO: How do we define access? Secrets?
+}
+
+type GitHubRepositoryConfig struct {
+	// The owner of the repository (e.g. example in `example/test` or `https://github.com/example/test`).
+	Owner string `json:"owner,omitempty"`
+	// The name of the repository (e.g. test in `example/test` or `https://github.com/example/test`).
+	Repository string `json:"repository,omitempty"`
+
+	// TODO: Do we want an SSH url instead maybe?
+	// TODO: On-prem GitHub Enterprise support?
+	// TODO: How do we define access? Secrets?
+
+	// Whether we should commit to change branches and use a Pull Request flow to achieve this.
+	// By default, this is false (i.e. we will commit straight to the main branch).
+	BranchWorkflow bool `json:"branchWorkflow,omitempty"`
+
+	// Whether we should show dashboard previews in the pull requests caused by the BranchWorkflow option.
+	// By default, this is false (i.e. we will not create previews).
+	// This option is a no-op if BranchWorkflow is `false` or default.
+	GenerateDashboardPreviews bool `json:"generateDashboardPreviews,omitempty"`
+}
```

Comment:
Yes. We'll be using a secret (from the Enterprise team), so it won't be a raw token here. I'm not quite sure how it'll look until they provide a stable API, but I _guess_ it will be a UUID reference to a secret in the secrets manager apiserver.